moveos/moveos-stdlib/moveos-stdlib/sources/object_storage.move

```diff
@@ -33,13 +34,31 @@ module moveos_std::object_storage {
     public fun remove<T: key>(this: &mut ObjectStorage, object_id: ObjectID): Object<T>{
         raw_table::remove<ObjectID, Object<T>>(*&this.handle, object_id)
     }
-    
+
+    const EObjectOwnerIsEmpty: u64 = 101;
     #[private_generic(T)]
     /// Add object to object store
-    public fun add<T: key>(this: &mut ObjectStorage, obj: Object<T>) {
+    public(friend) fun add<T: key>(this: &mut ObjectStorage, obj: Object<T>) {
+        assert!(option::is_some(&object::owner(&obj)), EObjectOwnerIsEmpty);
         raw_table::add<ObjectID, Object<T>>(*&this.handle, object::id(&obj), obj);
     }
 
+    #[private_generic(T)]
+    public fun transfer<T: key>(this: &mut ObjectStorage, obj: Object<T>, recipient: address) {
```

Comment:
I think the `transfer`,`freeze_object`,`share_object` should be a function in object module

```move
public fun transfer<T:key>(this: &mut Object<T>, recipient: address)
```